src/components/textarea/textarea.tsx

```diff
@@ -0,0 +1,498 @@
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  onFormReset(): void {
+    this.value = this.defaultValue;
+  }
+
+  onLabelClick(): void {
+    this.setFocus();
+  }
+
+  handleInput = (event: InputEvent): void => {
+    this.value = event.target["value"];
+    this.calciteTextareaInput.emit();
+  };
+
+  handleChange = (): void => {
+    this.calciteTextareaChange.emit();
+  };
+
+  footerEndSlotChangeHandler = (event: Event): void => {
+    this.endSlotElements = slotChangeGetAssignedElements(event);
+    if (this.disabled) {
+      this.disablePointerEvents(this.disabled, this.endSlotElements);
+    }
+  };
+
+  footerStartSlotChangeHandler = (event: Event): void => {
+    this.startSlotElements = slotChangeGetAssignedElements(event);
+    if (this.disabled) {
+      this.disablePointerEvents(this.disabled, this.startSlotElements);
+    }
+  };
+
+  contentSlotChangeHandler = (): void => {
+    if (!this.value) {
+      const nodes = this.el.childNodes;
+      nodes.forEach((el) => {
+        if (el.nodeName === "#text") {
+          this.value = el.nodeValue.trim();
+        }
+      });
+    }
+  };
+
+  renderCharacterLimit = (): VNode => {
+    return this.maxLength ? (
+      <span class={CSS.characterLimit}>
+        <span class={{ [CSS.characterOverlimit]: this.value?.length > this.maxLength }}>
+          {this.getLocalizedCharacterLength()}
+        </span>
+        {"/"}
+        {numberStringFormatter.localize(this.maxLength.toString())}
+      </span>
+    ) : null;
+  };
+
+  getLocalizedCharacterLength(): string {
+    numberStringFormatter.numberFormatOptions = {
+      locale: this.effectiveLocale,
+      numberingSystem: this.numberingSystem,
+      signDisplay: "never",
+      useGrouping: this.groupSeparator
+    };
+    return numberStringFormatter.localize(this.value ? this.value.length.toString() : "0");
+  }
+
+  disablePointerEvents(disabled: boolean, slottedElements: Element[]): void {
+    if (!!slottedElements?.length) {
+      slottedElements.forEach((el: HTMLElement) => {
+        el.style.pointerEvents = disabled ? "none" : "auto";
+      });
+    }
+  }
+
+  resizeObserver = createObserver("resize", () => {
+    const { textareaHeight, textareaWidth, elHeight, elWidth, footerHeight, footerWidth } =
+      this.getHeightandWidthOfElements();
+    if (footerWidth > 0 && footerWidth !== textareaWidth) {
+      this.footerEl.style.width = `${textareaWidth}px`;
+    }
+    if (elWidth !== textareaWidth || elHeight !== textareaHeight + (footerHeight || 0)) {
+      this.setHeightAndWidthToAuto();
+    }
+  });
+
+  syncHiddenFormInput(input: HTMLInputElement): void {
+    input.setCustomValidity("");
+    if (this.value?.length > this.maxLength) {
+      input.setCustomValidity(this.messages.overLimit);
+    }
+  }
+
+  setHeightAndWidthToAuto = throttle(
+    (): void => {
+      this.verticalResizeDisabled || (this.el.style.height = "auto");
```

Comment:
This doesn't seem to be that performance intensive. Was this being throttled to batch style updates? If so, I think it'd be better to use `writeTask` for this.

Reply:
This is being throttle to avoid flickering of drag handle when the user drags quick where and the height and width not set to `auto` yet. This is inconsistent and with throttling we are delaying it by 100ms which provides a better experience. And also subsequent tasks will be batched and executed.